modin/pandas/io.py

```diff
@@ -219,7 +220,11 @@ def read_excel(
 ):
     _, _, _, kwargs = inspect.getargvalues(inspect.currentframe())
     kwargs.update(kwargs.pop("kwds", {}))
-    return DataFrame(query_compiler=BaseFactory.read_excel(**kwargs))
+    parsed = BaseFactory.read_excel(**kwargs)
+    if isinstance(parsed, OrderedDict):
+        return parsed
```

Comment:
Same comment here for converting `pandas.DataFrame` to `modin.pandas.DataFrame`.